File: model/src/main/java/no/nav/pensjon/vtp/testmodell/repo/TestscenarioService.java
package no.nav.pensjon.vtp.testmodell.repo;

import java.util.Map;
import java.util.Optional;
import java.util.stream.Stream;

public interface TestscenarioService {
    Stream<Testscenario> findAll();

    Optional<Testscenario> getTestscenario(String id);

    Testscenario opprettTestscenario(TestscenarioTemplate template);

    Testscenario opprettTestscenario(TestscenarioTemplate template, Map<String, String> userSuppliedVariables);

    Testscenario opprettTestscenarioFraJsonString(String testscenarioJson, Map<String, String> userSuppliedVariables);

    void slettScenario(String id);
}

File: model/src/main/java/no/nav/pensjon/vtp/testmodell/repo/TestscenarioService.kt
package no.nav.pensjon.vtp.testmodell.repo

interface TestscenarioService {
    fun findAll(): List<Testscenario>
    fun getTestscenario(id: String): Testscenario?
    fun opprettTestscenario(template: TestscenarioTemplate): Testscenario
    fun opprettTestscenario(template: TestscenarioTemplate, userSuppliedVariables: Map<String, String> = mapOf()): Testscenario
    fun opprettTestscenarioFraJsonString(testscenarioJson: String, userSuppliedVariables: Map<String, String> = mapOf()): Testscenario
    fun slettScenario(id: String)
}
